### PR TITLE
RA-1004: Done some filtrations in the UI layer to improve role manage…

### PR DESCRIPTION
**Pull request by samuelmale**
…ment on manageRoles.page in reference Application

Ticket: https://issues.openmrs.org/browse/RA-1004

### CODE REVIEW

File: omod/src/main/webapp/resources/scripts/metadata/manageRoles.js [hunk 1]
```diff
@@ -122,12 +122,27 @@ angular.module("manageRoles", [ "roleService", "privilegeService", "ngDialog", "
                         $scope.privileges = results;
                         $scope.inheritedPrivilegeFlags = Array($scope.privileges.length);
                         $scope.privilegeFlags = Array($scope.privileges.length);
-                                                
+                        filterPrivileges(); //Do some filtration
```
Comment (dkayiwa) [inline omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: Is this comment necessary?

Comment (samuelmale) [reply omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: Ooohhh, let me remove it immediately

File: omod/src/main/webapp/resources/scripts/metadata/manageRoles.js [hunk 2]
```diff
@@ -122,12 +122,27 @@ angular.module("manageRoles", [ "roleService", "privilegeService", "ngDialog", "
...
+            function filterPrivileges(){
+            	var inheritedPrivileges =[];
+            	$scope.privileges.forEach(function(val, idx) {
+            		if($scope.privileges[idx].name.indexOf(":")>-1){
+            		var privType = $scope.privileges[idx].name.slice(0,$scope.privileges[idx].name.indexOf(":"));
+            		if(privType == "App"|| privType == "Task"){
+            			inheritedPrivileges.push(val);
+            		}}
+                });   
+            	$scope.privileges = [];//Empty privileges list
```
Comment (dkayiwa) [inline omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: As for some comments in your code, take a look at this: https://dzone.com/articles/5-best-practices-commenting

Comment (samuelmale) [reply omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: Hahaha I have looked through the resource. Obvious comments aren't required. You know there is this Open Source book I read before were they over advised commenting code for other devs to follow. Note taking in note that not every line code needs a comment. Thanks @dkayiwa . I will become a better me soon..

Comment (dkayiwa) [reply omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: Why don't you just make your code self documenting with well named variables, methods, and other goodies? Definitely if your comments are teaching someone a programming language, they are misplaced. Such belong to computer science text books. 😊 

Commenting should be reserved only for those rare cases where it is necessary. 😊

**Comment by samuelmale** [comment]
@dkayiwa  I removed useless comments

File: omod/src/main/webapp/resources/scripts/metadata/manageRoles.js (hunk 2 above)
Comment (samuelmale) [reply omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: Hahaha okay thanks. @dkayiwa

File: omod/src/main/webapp/resources/scripts/metadata/manageRoles.js [hunk 3]
```diff
@@ -122,12 +122,27 @@ angular.module("manageRoles", [ "roleService", "privilegeService", "ngDialog", "
...
                         loadInheritedPrivileges();
 
                     });
                 });
             }
+            //Filters Privileges to required criteria. ie Only "Application privileges" are required.
+            function filterPrivileges(){
+            	var inheritedPrivileges =[];
+            	$scope.privileges.forEach(function(val, idx) {
+            		if($scope.privileges[idx].name.indexOf(":")>-1){
```
Comment (wluyima) [inline omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: Don't you only need to check if the privilege has a **Privilege:** prefix?

Comment (samuelmale) [reply omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: @wluyima  you know am kinda of a newbie, but I thought this checks for presence of the prefix ":"
```
if($scope.privileges[idx].name.indexOf(":")>-1){
```
Or I was required to do it in a cleaner way?

Comment (samuelmale) [reply omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: Or did you mean checking  for presence of prefix before I run the entire function?
I didn't get you properly mentor!

**Comment by samuelmale** [comment]
@dkayiwa  done more indention .

File: omod/src/main/webapp/resources/scripts/metadata/manageRoles.js [hunk 4]
```diff
@@ -122,12 +122,28 @@ angular.module("manageRoles", [ "roleService", "privilegeService", "ngDialog", "
...
                     });
                 });
             }
+            //Filters Privileges to required criteria. ie Only "Application privileges" are required.
+            function filterPrivileges() {
+            	var inheritedPrivileges =[];
+            	$scope.privileges.forEach(function(val, idx) {
+            		if ($scope.privileges[idx].name.indexOf(":") > -1) {
+            		   var privType = $scope.privileges[idx].name.slice(0,$scope.privileges[idx].name.indexOf(":"));
+            		   if (privType == "App"|| privType == "Task") {
```
Comment (wluyima) [inline omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: Ticket says should only inherit privilege level roles

File: omod/src/main/webapp/resources/scripts/metadata/manageRoles.js [hunk 5]
```diff
@@ -122,12 +122,28 @@ angular.module("manageRoles", [ "roleService", "privilegeService", "ngDialog", "
...
-                                                
+                        filterPrivileges();                       
                         loadInheritedPrivileges();
 
                     });
                 });
             }
+            //Filters Privileges to required criteria. ie Only "Application privileges" are required.
+            function filterPrivileges() {
+            	var inheritedPrivileges =[];
```
Comment (wluyima) [inline omod/src/main/webapp/resources/scripts/metadata/manageRoles.js]: I think you are using the wrong name here, these aren't inherited privileges, they are application privileges

**Pull request merged**